manila/share/drivers/vastdata/__init__.py

```diff
@@ -72,73 +71,60 @@
         'vast_mgmt_password',
         help='Password for VAST management',
         secret=True),
+    cfg.BoolOpt(
+        'vast_dont_use_trash_api',
+        help='Do not use VAST trash API',
+        default=False,
+    )
 ]
 
 CONF = cfg.CONF
 CONF.register_opts(OPTS)
 
-_MANILA_TO_VAST_ACCESS_LEVEL = {
+MANILA_TO_VAST_ACCESS_LEVEL = {
     constants.ACCESS_LEVEL_RW: 'nfs_read_write',
     constants.ACCESS_LEVEL_RO: 'nfs_read_only',
 }
 
-RE_IS_IP = re.compile(r"\d+\.\d+\.\d+\.\d+")
-
 
 class VASTShareDriver(driver.ShareDriver):
     VERSION = '1.0'  # driver version
 
     def __init__(self, *args, **kwargs):
-        super(VASTShareDriver, self).__init__(False, *args, **kwargs)
+        super().__init__(False, *args, **kwargs)
         self.configuration.append_config_values(OPTS)
 
     def do_setup(self, context):
         """Driver initialization"""
 
-        auth = (self.configuration.vast_mgmt_user, self.configuration.vast_mgmt_password)
-        self._host = self.configuration.vast_mgmt_host
-        self._vippool_name = self.configuration.vast_vippool_name
-        self._root_export = self.configuration.vast_root_export
-
-        self.vms_session = RESTSession(base_url="https://{}/api".format(self._host), auth=auth, ssl_verify=False)
-        try:
-            metrics_spec = self.vms_session.metrics()
-        except VastApiException as ex:
-            msg = _("Exception when logging into the array: %s\n") % ex
-            LOG.exception(msg)
-            raise exception.ManilaException(message=msg)
-
-        self._metrics_spec = {m.fqn: m for m in metrics_spec}
+        manila_view = "manila"
         backend_name = self.configuration.safe_get('share_backend_name')
         self._backend_name = backend_name or self.__class__.__name__
+        self._vippool_name = self.configuration.vast_vippool_name
+        self._root_export = "/" + self.configuration.vast_root_export.strip("/")
+        self._dont_use_trash_api = self.configuration.vast_dont_use_trash_api
 
-        manila_view = "manila"
-        export = self._get_export(manila_view)
-        if not export:
-            policy = self._get_policy("default")
-            data = dict(name=manila_view, path=self._root_export, policy_id=policy.id, create_dir=True,
-                        protocols=['NFS'])
-            self.vms_session.post("views", data=data)
+        username = self.configuration.vast_mgmt_user
+        password = self.configuration.vast_mgmt_password
+        host = self.configuration.vast_mgmt_host
+        self.rest = RestApi(host, username, password, False, self.VERSION)
 
-        LOG.debug('setup complete')
+        if not self.rest.views.one(name=manila_view):
+            policy = self.rest.view_policies.one(name="default")
+            self.rest.views.create(name=manila_view, path=self._root_export, policy_id=policy.id)
```

Comment:
any idea why we're doing this?...

Reply:
For local mounting of parent directory before deletion.
I deleted local mount in favor of using Trash API so this part is also deleted.